docs/node/network-migration.mdx

```diff
@@ -0,0 +1,100 @@
+
+# Generate app hash using sha256sum.
+shasum -a 256 /path/to/snapdir/kwildb-snapshot.sql
+4c2ecfc145c37a45a1fd21cd38ee84db80d852996fc30438a007ba624387c683  /path/to/snapdir/kwildb-snapshot.sql
+
+# app_hash in genesis file is a hexadecimal representation of the sha256sum hash encoded in base64.
+# hex-> binary -> base64
+echo -n "4c2ecfc145c37a45a1fd21cd38ee84db80d852996fc30438a007ba624387c683" | xxd -r -p | base64
+
+TC7PwUXDekWh/SHNOO6E24DYUplvwwQ4oAe6YkOHxoM=
+
+# compare the app hash with the genesis file
+cat /path/to/snapdir/genesis.json | jq .app_hash
+
+"TC7PwUXDekWh/SHNOO6E24DYUplvwwQ4oAe6YkOHxoM="
+```
+
+To start the new network with the previous data, the genesis file must have the `app_hash` field set. If the `app_hash` field is not set, the network will start with an empty state.
+
+### Distribute Genesis State
+
+Distribute the `snapshot` and the `genesis.json` files required to start the new network. 
+
+Ensure that the `genesis.json` file has the `app_hash` field set to the hash of the uncompressed snapshot file. Refer to the [Verify App Hash](#verify-app-hash) section to correctly generate this hash. The genesis file is already set to reflect the genesis validators set to the previous network's validators set around the time of the snapshot.  If changes are needed for the genesis validators set or other configurations, update the genesis file manually as required before distributing it.
+
+### Run Genesis Nodes
+
+#### Install Kwild Binaries
+
+Firstly, install the new version of the `kwild` binaries. Refer to the [Installation](/docs//ref/installation) documentation for more details on getting the required versions of the `kwild` binaries.
+
+#### Download Genesis State
+
+Download the `snapshot` and the `genesis.json` files required for this new network. 
+
+#### Configure Node to Use Snapshot
+
+Refer to the [Node Configuration](/docs//ref/configuration) documentation to configure the `config.toml` file.
+
+To initialize the node with the snapshot data, the `snapshot_file` path in the `config.toml` file must be set to the location of the downloaded snapshot file.  The node will fail to start, if the `snapshot_file` path is not provided in the `config.toml`.
+
+#### Start the Node
+
+Once the genesis and config files are configured to use snapshots as described in the previous section, start the node using the following command:
+
+```bash
+# use -h for more options. 
+kwild -r /path/to/node/root/dir
+```
+
+### Nodes Catching Up
+
+Follow the same steps described in the [Run Genesis Nodes](#run-genesis-nodes) section to configure and start the nodes joining the upgraded network. Nodes can either sync with the network using `statesync` or `blocksync`.
+
+If the node uses `blocksync`, the `snapshot_file` path is mandatory to ensure that the node initializes with the same genesis initial state and replay all the historical blocks of the chain to catchup with the network without forking itself off the chain. 
```

Comment:
This is a bit of a run on sentence. Do the changes below work:

```suggestion
If the node uses `blocksync`, the `snapshot_file` path is mandatory to ensure that the node initializes with the same genesis initial state and replay all the historical blocks. If the node is not able to replay the historical blocks, it will fork itself off the chain. 
```